what4/src/What4/Expr/Builder.hs

```diff
@@ -334,6 +345,53 @@ data NonceApp t (e :: BaseType -> Type) (tp :: BaseType) where
         -> !(Ctx.Assignment e args)
         -> NonceApp t e ret
 
+
+-------------------------------------------------------------------------------
+-- BVOrSet
+
+data BVOrNote w = BVOrNote !IncrHash !(BVD.BVDomain w)
```

Comment:
It would be nice to use a special bit-oriented abstract domain here instead of the normal contiguous-range-of-bitvectors one. But it probably makes sense to do that as a separate PR.

Reply:
Agreed, on both counts.